wpilibc/src/main/native/cpp/DriverStation.cpp

```diff
@@ -26,41 +28,70 @@
 #include "frc/WPIErrors.h"
 
 namespace frc {
+// A simple class which caches the previous value written to an NT entry
+// Used to prevent redundant, repeated writes of the same value
+template <class T>
+class MatchDataSenderEntry {
+ public:
+  MatchDataSenderEntry(std::shared_ptr<nt::NetworkTable> table,
+                       const std::string& key, const T& initialVal) {
```

Comment:
Do `const std::shared_ptr<nt::NetworkTable>& table` and `const char* key` to avoid some extraneous copies/constructions.

Reply:
Updated the table, and changed the std::string to a wpi::Twine